sklearn/gaussian_process/kernels.py

```diff
@@ -0,0 +1,475 @@
+""" Kernels for Gaussian process regression and classification.
+
+The kernels in this module allow kernel-engineering, i.e., they can be
+combined via the "+" and "*" operators. These expressions can also contain
+scalar values, which are automatically converted to a constant kernel.
+
+All kernel allow (analytic) gradient-based hyperparameter optimization.
+The space of hyperparameters can be specified by giving lower und upper
+boundaries for the value of each hyperparameter (the search space is thus
+rectangular). This can be achieved by using a pair or triple instead of a
+single float wherever a parameter value is specified. In case of a pair,
+the first value specifies the lower boundary and the second value the upper
+boundary. In case of a triple, the middle value specified the initial value
+of the parameter during hyperparameter-optimization.
```

Comment:
Are the bounds really important in practice? The reason I am asking is because this reduces the number of solvers we can use. In SciPy, there is L-BFGS-B. We can also use the projected gradient method (gradient descent followed by a truncation).